trade-finance-manager-ui/server/controllers/case/amendments/amendmentFacilityEndDate.get.test.js

```diff
@@ -0,0 +1,269 @@
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE });
+      api.getFacility = jest.fn().mockResolvedValueOnce(bssEwcsFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+
+      await getAmendmentFacilityEndDate(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/amendment-options`);
+    });
+
+    it("should redirect to the 'Has the bank provided a facility end date' question page when the user has not yet selected if the facility end date is being used", async () => {
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: { ...MOCK_AMENDMENT_COVERENDDATE_CHANGE, isUsingFacilityEndDate: undefined } });
+      api.getFacility = jest.fn().mockResolvedValueOnce(gefFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+
+      await getAmendmentFacilityEndDate(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/is-using-facility-end-date`);
+    });
+
+    it("should redirect to the 'Has the bank provided a facility end date' page if not using facility end date", async () => {
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: { ...MOCK_AMENDMENT_COVERENDDATE_CHANGE, isUsingFacilityEndDate: false } });
+      api.getFacility = jest.fn().mockResolvedValueOnce(gefFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+
+      await getAmendmentFacilityEndDate(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/is-using-facility-end-date`);
+    });
+
+    it('should render the template with isEditable true when the amendment is found and is in progress', async () => {
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE_USING_FACILITY_ENDDATE });
+      api.getFacility = jest.fn().mockResolvedValueOnce(gefFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+      await getAmendmentFacilityEndDate(req, res);
+
+      expect(res.render).toHaveBeenCalledWith('case/amendments/amendment-facility-end-date.njk', {
+        dealId,
+        facilityId,
+        dayInput: '',
+        monthInput: '',
+        yearInput: '',
+        isEditable: true,
+        user,
+      });
+    });
+
+    it('should render the template with isEditable false when amendment is found but has been completed', async () => {
```

Comment:
the it statements are quite long in this file, can you add describes around them please

Reply:
Also if using describes, can put a beforeEach in each describe so the it blocks are cleaner.  Can do let req at the top of the page